apps/shipments/serializers.py

```diff
@@ -177,6 +178,8 @@ def update(self, instance, validated_data):
         if 'device' in self.context:
             if validated_data['device_id']:
                 instance.device = self.context['device']
+                device = Device.get_or_create_with_permission(self.context['auth'], validated_data.pop('device_id'))
```

Comment:
I'm confused by this change - is it necessary?

Reply:
This is probable an error that I carried on during the various `rebase`s . Good catch :+1: